src/services/engine_manager.cpp

```diff
@@ -111,9 +114,21 @@ namespace otto::services {
         case ScreenEnum::synth: [[fallthrough]];
         case ScreenEnum::synth_envelope: [[fallthrough]];
         case ScreenEnum::synth_selector: [[fallthrough]];
-        case ScreenEnum::voices:
-          if (ui_manager.state.active_channel != +ChannelEnum::internal)
+        case ScreenEnum::voices: {
+          if (ui_manager.state.active_channel != +ChannelEnum::internal) {
             ui_manager.state.active_channel = +ChannelEnum::internal;
+            break;
+          }
+        }
+        case ScreenEnum::external: {
+          if (line_in.props.mode.get() == 2 && line_in.props.active_send.get() == 0)
```

Comment:
move this logic into `ChannelEnum External::channel()` or similar

Reply:
Good idea.